src/components/Dropdown/Dropdown.tsx

```diff
@@ -74,7 +74,7 @@ type DropdownProps = ListboxProps & {
   optionsClassName?: string;
 };
 
-type RenderProp<Arg> = (arg: Arg) => ReactNode;
+type RenderProp<Arg> = (arg: Arg) => ReactElement;
```

Comment:
I assume some of these changes are b/c of React 18 being slightly different or more strict?

Also in this case `children` is either a `ReactNode` or `ReactElement`. So when we cast on ln285, would this cause issues? is one of these a super-set of the other?